src/main/java/fi/otavanopisto/kuntaapi/server/integrations/casem/CaseMCacheUpdater.java

```diff
@@ -0,0 +1,839 @@
+    for (Node node : nodes) {
+      List<Long> childCaseMParentIds = new ArrayList<>(caseMParentIds);
+      childCaseMParentIds.add(node.getNodeId());
+      
+      if (node.getType().equals(0l)) {
+        // Type 0 pages are hidden levels that should be flatted out
+      } else {
+        Page page = translateNode(organizationId, caseMRootNodeId, parentNode, node);
+        caseMCache.cacheNode(organizationId, page);
+      }
+      
+      cacheNodeTree(organizationId, caseMRootNodeId, node, getChildNodes(organizationId, caseMRootNodeId, childCaseMParentIds), childCaseMParentIds);
+    }
+  }
+  
+  private Long getCaseMRootNodeId(OrganizationId organizationId) {
+    String rootNode = organizationSettingController.getSettingValue(organizationId, CaseMConsts.ORGANIZATION_SETTING_ROOT_NODE);
+    if (StringUtils.isNumeric(rootNode)) {
+      return NumberUtils.createLong(rootNode);
+    }
+    
+   return null;
+  }
+  
+  private String getCaseMDownloadUrl(OrganizationId organizationId) {
+    String baseUrl = organizationSettingController.getSettingValue(organizationId, CaseMConsts.ORGANIZATION_SETTING_BASEURL);
+    String path = organizationSettingController.getSettingValue(organizationId, CaseMConsts.ORGANIZATION_SETTING_DOWNLOAD_PATH);
+    return String.format("%s/%s", baseUrl, path);
+  }
+
+  private List<Node> getChildNodes(OrganizationId organizationId, Long caseMRootNodeId, List<Long> caseMParentIds) {
+    List<Node> result = new ArrayList<>();
+    Long skipToken = null;
+    
+    do {
+      NodeList nodeList = getChildNodeList(organizationId, caseMRootNodeId, caseMParentIds, skipToken);
+      if (nodeList == null) {
+        break;
+      }
+      
+      result.addAll(nodeList.getValue());
+      
+      skipToken = getSkipToken(nodeList.getOdataNextLink());
+    } while (skipToken != null);
+    
+    Collections.sort(result, new NodeComparator());
+    
+    return result;
+  }
+  
+  private NodeList getChildNodeList(OrganizationId organizationId, Long caseMRootNodeId, List<Long> caseMParentIds, Long skipToken) {
+    String pathQuery = getSubNodePath(caseMParentIds);
+    
+    ApiResponse<NodeList> response = caseMApi.getNodesApi(organizationId)
+      .listSubNodes(caseMRootNodeId, pathQuery, skipToken != null ? String.valueOf(skipToken) : null);
+    
+    if (!response.isOk()) {
+      logger.severe(String.format("Listing nodes by rootNode %d and pathQuery %s failed on [%d] %s", caseMRootNodeId, pathQuery, response.getStatus(), response.getMessage()));
+      return null;
+    } else {
+      return response.getResponse();
+    }
+  }
+  
+  private String getSubNodePath(List<Long> caseMParentIds) {
+    StringBuilder result = new StringBuilder();
+    
+    for (Long caseMParentId : caseMParentIds) {
+      result.append(String.format("SubNodes(%d)/", caseMParentId));
+    }
+    
+    result.append("SubNodes()");
+    
+    return result.toString();
+  }
+  
+  private Long getSkipToken(String nextLink) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this unused private "getSkipToken" method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AUnusedPrivateMethod)